syntax = "proto2";

package b0.core_msgs;

// A message envelope used to wrap (optionally: compress) the real message payload
message MessageEnvelope {
  required bytes payload = 1; // payload
  required int32 uncompressed_size = 2; // size of the uncompressed payload
  required string compression_algorithm = 3; // compression algorithm used, or blank if no compression
  required string type = 4; // a string indicating the type of the payload
}

